flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -335,6 +380,34 @@ public void checkpointState(
         }
     }
 
+    private void registerAlignmentTimer(
+            long checkpointId,
+            OperatorChain<?, ?> operatorChain,
+            CheckpointBarrier checkpointBarrier) {
+        if (alignmentTimer != null) {
```

Comment:
shouldn't this be `checkState(alignmentTimer == null)`?

Reply:
Actually, it may be null.

For a job without back pressure, the CP is completed quickly, but the timer is not triggered, so the next time the CP will continue, so this may be null. It should be reasonable to cancel the previous timer here.

Reply:
I see that in the end you have added the `checkState(alignmentTimer == null)` that I suggested?

Reply:
Yes, I found some unit tests failed after added the `checkState(alignmentTimer == null)`. I have changed in my Mac, I will re-submit commits after everything  is ok.